src/svea_sensors/scripts/wheel_encoder_reader.py

```diff
@@ -30,14 +30,15 @@
 
 def set_covariance(cov_matrix, linear_covariance, angular_covariance):
     """Setup the covariance matrix of the twist message"""
-    cov_step = 7 # Index step in covariance matrix
-    lin_cov_end = cov_step + 1
-    for i in range(0, lin_cov_end, cov_step):
-        cov_matrix[i] = linear_covariance
-    ang_cov_start = lin_cov_end + cov_step - 1
-    ang_cov_end = lin_cov_end + 3*cov_step
-    for i in range(ang_cov_start, ang_cov_end, cov_step):
-        cov_matrix[i] = angular_covariance
+    cov_step = 7
```

Comment:
Ah, I see why this is necessary, we update the content of `cov_matrix`.  This is not very good SE, is it possible to change where `set_covariance` is called so that it returns a new matrix instead of mutating `cov_matrix`?